### PR TITLE
Adding Fluorine as a new element in RMG

### PR DESCRIPTION
**Pull request by oscarwumit**
### Motivation or Problem
Adding Fluorine as an atomtype to RMG.

### Description of Changes
Files related to atomtypes are changed.

### Testing
Added Fluorine atomtype test.

### CODE REVIEW

File: rmgpy/reaction.py [hunk 1]
```diff
@@ -964,14 +964,16 @@ def generatePairs(self):
...
-            productIodines    = [sum([1 for atom in  product.molecule[0].atoms if atom.isChlorine()]) for product  in products ]
+            reactantIodines   = [sum([1 for atom in reactant.molecule[0].atoms if atom.isIodine()]) for reactant in reactants]
+            productIodines    = [sum([1 for atom in  product.molecule[0].atoms if atom.isIodine()]) for product  in products ]
+            reactantFluorines = [sum([1 for atom in reactant.molecule[0].atoms if atom.isFluorine()]) for reactant in reactants]
+            productFluorines  = [sum([1 for atom in  product.molecule[0].atoms if atom.isFluorine()]) for product  in products ]
             
             # Sort the reactants and products by C/O/N/S numbers
-            reactants = [(carbon, oxygen, nitrogen, silicon, sulfur, chlorine, iodine, reactant) for carbon, oxygen, nitrogen, silicon, sulfur, chlorine, iodine, reactant
+            reactants = [(carbon, oxygen, nitrogen, silicon, sulfur, chlorine, iodine, fluorine, reactant) for carbon, oxygen, nitrogen, silicon, sulfur, chlorine, iodine, reactant
```
Comment (alongd) [inline rmgpy/reaction.py]: I think this should be:
```suggestion
            reactants = [(carbon, oxygen, nitrogen, silicon, sulfur, chlorine, iodine, fluorine, reactant) for carbon, oxygen, nitrogen, silicon, sulfur, chlorine, iodine, fluorine, reactant
```
(same for `products` below)

File: rmgpy/reaction.py [hunk 2]
```diff
@@ -964,14 +964,16 @@ def generatePairs(self):
...
                          in zip(reactantCarbons,reactantOxygens,reactantNitrogens,reactantSilicons,reactantSulfurs,reactantChlorines, reactantIodines, reactants)]
```
Comment (alongd) [inline rmgpy/reaction.py]: I think `reactantFluorines` should be added to the zip here, same for products below

File: rmgpy/reaction.py [hunk 3]
```diff
@@ -964,14 +964,16 @@ def generatePairs(self):
             productSulfurs    = [sum([1 for atom in  product.molecule[0].atoms if atom.isSulfur()])   for product  in products ]
             reactantChlorines = [sum([1 for atom in reactant.molecule[0].atoms if atom.isChlorine()]) for reactant in reactants]
             productChlorines  = [sum([1 for atom in  product.molecule[0].atoms if atom.isChlorine()]) for product  in products ]
-            reactantIodines   = [sum([1 for atom in reactant.molecule[0].atoms if atom.isChlorine()]) for reactant in reactants]
-            productIodines    = [sum([1 for atom in  product.molecule[0].atoms if atom.isChlorine()]) for product  in products ]
+            reactantIodines   = [sum([1 for atom in reactant.molecule[0].atoms if atom.isIodine()]) for reactant in reactants]
+            productIodines    = [sum([1 for atom in  product.molecule[0].atoms if atom.isIodine()]) for product  in products ]
+            reactantFluorines = [sum([1 for atom in reactant.molecule[0].atoms if atom.isFluorine()]) for reactant in reactants]
+            productFluorines  = [sum([1 for atom in  product.molecule[0].atoms if atom.isFluorine()]) for product  in products ]
```
Comment (alongd) [inline rmgpy/reaction.py]: Do we have the `isFluorine()` method already defined?

Comment (oscarwumit) [reply rmgpy/reaction.py]: Which file should I modify to define this function?

Comment (alongd) [reply rmgpy/reaction.py]: We do already have `isFluorine()`, it's [here](https://github.com/ReactionMechanismGenerator/RMG-Py/blob/c61d2ef40fbbc9cf34363103e934a39d77e063d2/rmgpy/molecule/molecule.py#L320)

**Pull request merged**